### PR TITLE
fix bug in ModifiedSphericalBessel

### LINKED ISSUE #222: Double free / corruption with DVR
Running the python script below with the latest master gives "corrupted size vs. prev_size" or segfaults on python exit.

```py
#!/usr/bin/env python3
import numpy as np
from ase import Atoms

import rascal
from rascal.representations import SphericalInvariants as SOAP

HYPERS = {
    'soap_type': 'RadialSpectrum',
    'interaction_cutoff': 3.5,
    'max_radial': 15,
    'max_angular': 0,
    'gaussian_sigma_constant': 0.5,
    'gaussian_sigma_type': 'Constant',
    'cutoff_smooth_width': 0.5,
    'radial_basis': 'DVR',
}

h4 = Atoms(
    symbols='H5',
    positions=[(0, 0, 0), (0, 0, 2), (0, 0, -2), (0, 2, 0), (0, -2, 0)],
    cell=[20, 20, 20],
    pbc=True
)
h4.wrap()

soap = SOAP(**HYPERS)
feat = soap.transform(h4)

feat = feat.get_features(soap)
```

Am I doing something wrong or is this a memory error?

### PR DESCRIPTION
**Pull request by felixmusil**
Fix #222 by handling properly the special case l_max = 0 in bessel.hh

ModifiedSphericalBessel is now tested using l_max = 0 against mpmath reference. 

Tasks before review:

- [ x] Add tests, examples, and complete documentation of any added functionality
    - [x ] Make sure the examples run (!) and produce the expected output
    - [ x] For bugfix pull requests, list here which tests have been added or re-enabled
          to verify the fix and catch future regressions as well as similar bugs
          elsewhere in the code.
- [ x] Run `make lint` on the project, ensure it passes
    - [x ] Run `make pretty-cpp` and `make pretty-python`, check that the
          auto-formatting is sensible

### CODE REVIEW

**Comment by Luthaf** [comment]
Should we also add some basic tests of DVR with different calculators?

**Comment by felixmusil** [comment]
when the gradients are there it will be easier. Also the time for the tests will increase.

**Comment by max-veit** [comment]
I see you changed the reference data file to the binary JSON format - in principle, the file is 50% smaller, but wouldn't the diff itself have been much smaller if you'd just stuck to the text-based format that Git can understand (and that we can view and edit)? Especially if you're not changing _all_ the reference data values, I think the text JSON might end up giving a smaller repo size in the end.

**Comment by felixmusil** [comment]
> I see you changed the reference data file to the binary JSON format - in principle, the file is 50% smaller, but wouldn't the diff itself have been much smaller if you'd just stuck to the text-based format that Git can understand (and that we can view and edit)? Especially if you're not changing all the reference data values, I think the text JSON might end up giving a smaller repo size in the end.

In principle these files should be in ubjson to save space so I am fixing this here. 
Also we are talking about 0.5 mb right ? so does it really matter considering the size of the repo as of now ?

**Comment by max-veit** [comment]
> In principle these files should be in ubjson to save space so I am fixing this here.

But what I'm saying is I don't understand _why_ they should be ubjson format. Since they seem to get updated much more frequently than the actual reference data they contain (because of trivial changes like parameter names), the space savings is quickly wiped out. Same goes here: There _is_ no space savings by switching to ubjson here; on the contrary, it adds 0.5 MiB to the repo's history (unless you rebase the branch and delete the old json file -- which, if it's in master, won't even work).

> Also we are talking about 0.5 mb right ? so does it really matter considering the size of the repo as of now ?

That sort of argument is how we got to the current repo bloat situation. A megabyte isn't _small_ for most git repos, and all these little space inefficiencies add up over time.

**Comment by felixmusil** [comment]
> That sort of argument is how we got to the current repo bloat situation.

I don't think that this is true. The main problem for the size of the repo comes from the overall design of the test framework. We don't have a solution for it so I would rather not have to bother about something that does not matter anyway. 
Also a sizable part of the file has been removed, i.e. the file has to be modified.

> A megabyte isn't small for most git repos

it is for ours, for better or worse.

Could you review the actual purpose of this PR please ?

**Comment by Luthaf** [comment]
> when the gradients are there it will be easier but I would test it only in the radial integral test. So I would not do it here.

Do you mean w should add tests as part of #225 ?

**Comment by felixmusil** [comment]
> Do you mean w should add tests as part of #225 ?

the tests you are mentioning are part of this PR yes.

File: scripts/generate_modified_bessel_first_kind_ref_data.py [hunk 1]
```diff
@@ -9,17 +9,6 @@
...
-# and rescales them.
-
-
-def get_leggauss(order, a, b):
-    x, w = leggauss(order)
-    # rescaling
-    x = (b-a)*0.5 * x + 0.5*(a+b)
-    w = (b-a)*0.5 * w
-    return x, w
-
```
[the remark this answers is not in the document]
Comment (felixmusil) [reply scripts/generate_modified_bessel_first_kind_ref_data.py]: it is unused so I removed it. I will resolve the conflict when merging with master if problems arise.

**Pull request merged**